src/evo/deterministicmns.cpp

```diff
@@ -511,10 +513,28 @@ void CDeterministicMNList::UpdateMN(const CDeterministicMN& oldDmn, const std::s
     // Using this temporary map as a checkpoint to roll back to in case of any issues.
     decltype(mnUniquePropertyMap) mnUniquePropertyMapSaved = mnUniquePropertyMap;
 
-    if (!UpdateUniqueProperty(*dmn, oldState->addr, pdmnState->addr)) {
+    const auto updateNetInfo = [&]() {
+        if (oldState->netInfo != pdmnState->netInfo) {
+            // We track each individual entry in netInfo as opposed to netInfo itself (preventing us from
+            // using UpdateUniqueProperty()), so we need to successfully purge all old entries and insert
+            // new entries to successfully update.
+            for (const CService& old_entry : oldState->netInfo.GetEntries()) {
+                if (old_entry != CService() && !DeleteUniqueProperty(*dmn, old_entry)) {
+                    return strprintf("internal error"); // This shouldn't be possible
+                }
+            }
+            for (const CService& new_entry : pdmnState->netInfo.GetEntries()) {
+                if (new_entry != CService() && !AddUniqueProperty(*dmn, new_entry)) {
```

Comment:
same for `new_entry`

Reply:
Check was added to mirror implementation in `UpdateUniqueProperty()`  ([comment](https://github.com/dashpay/dash/pull/6627#discussion_r2076023182))